avaframe/com1DFA/deriveParameterSet.py

```diff
@@ -801,6 +799,94 @@ def checkRasterMeshSize(cfgSim, rasterFile, typeIndicator='DEM', onlySearch=Fals
 
     return pathToRaster
 
+def checkExtentAndCellSize(cfg, inputFile, dem, fileType):
+    """ check if extent of inputFile is within resizeThreshold of dem, if so resize and save to remeshedRasters
+
+        Parameters
+        -----------
+        cfg: configparser object
+            configuration settings
+        inputFile: pathlib Path
+            path to inputFile
+        dem: dict
+            dictionary with info on DEM
+        fileType: str
+            name of fileType
+    """
+
+    inputField = IOf.readRaster(inputFile)
+    cellSizeOld = inputField['header']['cellsize']
+    demHeader = dem['header']
+
+    if fileType == 'RELTH':
+        cT = 0.
+    else:
+        cT = float(cfg['GENERAL']['resizeThreshold'])
+
+    # compute difference of llcenter and urcenter - used for warning
+    diffX0 = inputField['header']['xllcenter'] - demHeader['xllcenter']
+    diffY0 = inputField['header']['yllcenter'] - demHeader['yllcenter']
+    diffX1 = ((inputField['header']['xllcenter']+ inputField['header']['ncols']* inputField['header']['cellsize']) -
+              (demHeader['xllcenter']+demHeader['ncols']*demHeader['cellsize']))
+    diffY1 = ((inputField['header']['yllcenter']+ inputField['header']['nrows']* inputField['header']['cellsize']) -
+              (demHeader['yllcenter']+demHeader['nrows']*demHeader['cellsize']))
+
+    if diffX0 > cT*demHeader['cellsize'] or diffY0 > cT*demHeader['cellsize']:
+        message = ('Lower left center coordinates of DEM and %s file are not within threshold %.2f x meshCellSize' %
+                   (fileType, cT))
+        log.error(message)
+        raise AssertionError(message)
+    elif diffX1 > cT*demHeader['cellsize'] or diffY1 > cT*demHeader['cellsize']:
+        message = ('Upper right center coordinates of DEM and %s file are not within threshold %.2f x meshCellSize' %
+                   (fileType, cT))
+        log.error(message)
+        raise AssertionError(message)
+    if fileType == 'RELTH':
+        if inputField['header']['cellsize'] != demHeader['cellsize']:
+            message = 'Cell size of RELTH file: %s does not match the desired computational meshCellSize' % inputFile.stem
+            log.error(message)
+            raise AssertionError(message)
+
+    # check if identical extent, if so use unchanged
+    if diffX0 == 0 and diffY0 == 0 and diffY1 == 0 and diffX1 == 0 and inputField['header']['cellsize'] == demHeader['cellsize']:
```

Comment:
Mh, we might have to use a threshold again, similar to meshCellSizeThreshold. We cannot guarantee that this always comes out to precisely 0...

Also you could use diffX0 == diffY0 == ... == 0 to shorten it a bit